File: backend/src/main/java/edu/cit/chan/unilost/dto/ItemRequest.java
package edu.cit.chan.unilost.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItemRequest {

    @NotBlank(message = "Title is required")
    @Size(max = 100, message = "Title must be 100 characters or less")
    private String title;

    @NotBlank(message = "Description is required")
    @Size(max = 1000, message = "Description must be 1000 characters or less")
    private String description;

    @NotBlank(message = "Type is required")
    @Pattern(regexp = "LOST|FOUND", message = "Type must be LOST or FOUND")
    private String type;

    @NotBlank(message = "Category is required")
    @Pattern(regexp = "ELECTRONICS|WALLETS|CLOTHING|DOCUMENTS|ACCESSORIES|BOOKS|KEYS|BAGS|OTHER",
            message = "Invalid category")
    private String category;

    @Size(max = 200, message = "Location must be 200 characters or less")
    private String location;

    @Size(max = 500, message = "Secret detail question must be 500 characters or less")
    private String secretDetailQuestion;

    private String dateLostFound;

    private String campusId;
}

File: mobile/app/src/main/java/com/hulampay/mobile/data/model/ItemRequest.kt
package com.hulampay.mobile.data.model

import com.google.gson.annotations.SerializedName

/**
 * Mirrors backend ItemRequest accepted by POST /api/items (multipart "item" part).
 *
 * Notes:
 * - `type` must be LOST or FOUND.
 * - `category` must be one of the backend enum names — see [ItemCategory].
 * - `dateLostFound` must be ISO-8601 LocalDateTime (e.g. "2026-05-13T00:00:00").
 * - If `campusId` is omitted the backend defaults to the reporter's universityTag.
 */
data class ItemRequest(
    @SerializedName("title")                val title: String,
    @SerializedName("description")          val description: String,
    @SerializedName("type")                 val type: String,
    @SerializedName("category")             val category: String,
    @SerializedName("location")             val location: String? = null,
    @SerializedName("latitude")             val latitude: Double? = null,
    @SerializedName("longitude")            val longitude: Double? = null,
    @SerializedName("secretDetailQuestion") val secretDetailQuestion: String? = null,
    @SerializedName("dateLostFound")        val dateLostFound: String? = null,
    @SerializedName("campusId")             val campusId: String? = null,
)

/**
 * Canonical backend category enum values. Display labels are matched by index.
 * Keep in sync with backend ItemRequest's @Pattern regex.
 */
object ItemCategory {
    val backendNames = listOf(
        "ELECTRONICS", "DOCUMENTS", "CLOTHING", "ACCESSORIES",
        "BOOKS", "BAGS", "KEYS", "WALLETS", "OTHER"
    )
    val displayLabels = listOf(
        "Electronics", "Documents", "Clothing", "Accessories",
        "Books", "Bags", "Keys", "Wallets", "Other"
    )

    fun displayToBackend(display: String): String? {
        val idx = displayLabels.indexOfFirst { it.equals(display, ignoreCase = true) }
        return if (idx >= 0) backendNames[idx] else null
    }
}
